python/mosaic/api/gdal.py

```diff
@@ -0,0 +1,59 @@
+from pyspark.sql import SparkSession
+from typing import Any
+
+__all__ = [
+    "setup_gdal",
+    "enable_gdal"
+]
+
+def setup_gdal(
+        spark: SparkSession,
+        init_script_path: str = "/dbfs/FileStore/geospatial/mosaic/gdal/",
+        shared_objects_path: str = "/dbfs/FileStore/geospatial/mosaic/gdal/") -> None:
+    """
+    Prepare GDAL init script and shared objects required for GDAL to run on spark.
+    This function will generate the init script that will install GDAL on each worker node.
+    After the setup_gdal is run, a cluster restart is required.
+
+    Parameters
+    ----------
+    spark : pyspark.sql.SparkSession
+            The active SparkSession.
+    init_script_path : str
+            Path to write out the init script for GDAL installation.
+    shared_objects_path : str
+            Path to write out shared objects (libgdalalljni.so and libgdalalljni.so.30) that GDAL requires at runtime.
+            Note: If you dont use the default path you will need to update the generated init script.
+
+    Returns
+    -------
+
+    """
+    sc = spark.sparkContext
+    mosaicContextClass = getattr(
+        sc._jvm.com.databricks.labs.mosaic.functions, "MosaicContext"
+    )
+    mosaicGDALObject = getattr(sc._jvm.com.databricks.labs.mosaic.gdal, "MosaicGDAL")
+    mosaicGDALObject.prepareEnvironment(spark._jsparkSession, init_script_path, shared_objects_path)
+
+
+
+def enable_gdal(spark: SparkSession) -> None:
```

Comment:
@edurdevic and @milos-colic Recommend there is a print statement that points to setup_gdal function if enable_gdal fails.

Reply:
it will throw the scala exception if it fails, which isnt great.
We can catch Py4j Exception but we are limited in what we can print there - "Something went wrong:" then pring scala exception? No sure it adds value

Reply:
I will print a "Success" statement though 
I think that helps.